templates/2024Github.html

```diff
@@ -0,0 +1,357 @@
+            </tr>
+        </thead>
+        <tbody>
+            {% for student in students %}
+            <tr>
+                <td><input type="checkbox" class="studentCheckbox"></td> <!-- Checkbox column -->
+                <td>{{ student[1] }}</td>
+                {% if student[11] and student[11] != '' %}
+                    <td><a href="mailto:{{ student[11] }}">{{ student[11] }}</a></td>
+                {% else %}
+                    <td></td>
+                {% endif %}
+                <td>{{ student[13] }}</td>
+                <td>{{ student[4] }}</td>
+                <td><a href="/view/{{ student[0] }}">View</a>&nbsp;<a href="/edit_student/{{ student[0] }}">Edit</a></td>
+            </tr>
+            {% endfor %}
+        </tbody>
+    </table>
+{% endblock %}
+
+
+{% block scripts %}
+
+    <script>
+        $(document).ready(function() {
+            var table = $('#studentTable').DataTable({
+                "lengthMenu": [50, 100, 250, 500],  // Set the available options for the number of entries
+                "dom": '<"top"if>rt<"bottom"lp><"clear">', // Place the information at the top
+            });
+
+            // Select/Deselect All checkboxes
+            $('#selectAllBtn').on('click', function() {
+                $('.studentCheckbox').prop('checked', true);
+                table.rows().select();
+            });
+
+            $('#deselectAllBtn').on('click', function() {
+                $('.studentCheckbox').prop('checked', false);
+                table.rows().deselect();
+            });
+
+            // Select/Deselect All checkboxes
+            $('#selectAllCheckbox').on('change', function() {
+                var isChecked = $(this).is(':checked');
+                $('.studentCheckbox').prop('checked', isChecked);
+                table.rows().select(isChecked);
+            });
+
+
+            // Download Key attributes button action
+            $('#downloadKeyAttributesBtn').on('click', function() {
+
+                var student_ids = [];
+                var selectedRows = $('.studentCheckbox:checked').each(function() {
+                    text = table.row($(this).closest('tr')).data();
+                    var id = text[1].replace(/<\/?[^>]+(>|$)/g, '')
+                    student_ids.push(id);
+                });
+
+                data = {
+                    student_ids: student_ids
+                };
+
+                // Construct the query string
+                var queryString = 'student_ids=' + student_ids.join(',');
+
+                url = '/download_key_attributes';
+
+                // Append the query string to the URL
+                url += '?' + queryString;
+
+                window.location.href = url;
+
+            });
+
+
+            // Copy Emails button action
+            $('#copyEmailsBtn').on('click', function() {
+                var emails = [];
+                var selectedRows = $('.studentCheckbox:checked').each(function() {
```

Comment:
Not working - copy WEHI email doesn't work and neither does COpy Personal.

I don't see how personal would ever work given there is no email for personal....